modules/utxo-lib/src/bitgo/tnumber.ts

```diff
@@ -0,0 +1,33 @@
+/** Convert input to bigint or number. Throws error if input cannot be converted to a safe integer number */
+export function toTNumber<TNumber extends number | bigint>(
+  value: number | bigint | string,
+  amountType: 'number' | 'bigint'
+): TNumber {
+  if (value === undefined) {
+    throw new Error();
+  }
+  if (amountType === 'number') {
+    return Number(value) as TNumber;
+  }
+  if (amountType === 'bigint') {
+    return BigInt(value) as TNumber;
+  }
+  throw new Error();
+}
+
+/** Scale a decimal amount (optionally to a bigint) */
+export function getValueScaled<TNumber extends number | bigint>(
+  value: number,
+  amountType: 'number' | 'bigint' = 'number',
+  scaleFactor = 1e8
+): TNumber {
```

Comment:
please add a test for this (could be a separate PR)

Reply:
Added